source/SH3/arc/vfile.cpp

```diff
@@ -81,9 +81,13 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
         e.set_error(load_result::PARTIAL_READ);
     }
 
-    assert((fpos + len) <= fsize); // This should never EVER happen. Terminate if it does.
+    assert(fpos <= fsize); // This should never EVER happen. Terminate if it does.
+
+    if((std::size_t nbytes = std::min(len, fsize - fpos)) == len)
```

Comment:
That should be `!= len`. Also I think this makes lines 79-82 redundant, as that case should be covered by this check.